src/puppetlabs/puppetdb/http/server.clj

```diff
@@ -52,15 +52,15 @@
 
   * `globals` - a map containing global state useful to request handlers.
 
-  * `authorized?` - a function that takes a request and returns a
-    truthy value if the request is authorized. If not supplied, we default
-    to authorizing all requests."
+  * `authorizer` - a function that takes a request and returns a
+    :authorized if the request is authorized, or a user-visible reason if not.
+    If not supplied, we default to authorizing all requests."
   [& options]
   (let [opts (apply hash-map options)]
     (-> (routes (get-in opts [:globals :url-prefix]))
         (wrap-resource "public")
         (wrap-params)
-        (wrap-with-authorization (opts :authorized? (constantly true)))
+        (wrap-with-authorization (opts :authorizer (constantly :authorized)))
```

Comment:
This may be out of the scope of this PR, but it seems like we could remove some duplication here by having this default in `wrap-with-authorization` itself. Also since we already have defaults here we don't really need them at the build app level.